python/cudf/cudf/core/column/column.py

```diff
@@ -1468,26 +1360,29 @@ def kernel_non_masked(values, results):
                 results[i] = core(values[i])
 
         kernel_non_masked.forall(len(column))(values, results)
-    # Output
-    return Buffer(results)
 
+    return as_column(results)
 
-def _data_from_cuda_array_interface_desc(desc):
+
+def _data_from_cuda_array_interface_desc(obj):
+    desc = obj.__cuda_array_interface__
     ptr = desc["data"][0]
     nelem = desc["shape"][0]
     dtype = np.dtype(desc["typestr"])
 
+    # TODO: this can be done more efficiently
     data = rmm.device_array_from_ptr(
         ptr, nelem=nelem, dtype=dtype, finalizer=None
     )
```

Comment:
Is this still used? If not, can it be dropped?

Reply:
Nice find!